homeassistant/components/enigma2/__init__.py

```diff
@@ -1 +1,44 @@
 """Support for Enigma2 devices."""
+
+import logging
+
+from openwebif.api import OpenWebIfDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = [Platform.MEDIA_PLAYER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Enigma2 from a config entry."""
+    device: OpenWebIfDevice = OpenWebIfDevice(
+        entry.options[CONF_HOST],
+        port=entry.options[CONF_PORT],
+        username=entry.options.get(CONF_USERNAME),
+        password=entry.options.get(CONF_PASSWORD),
+        is_https=entry.options.get(CONF_SSL),
+    )
```

Comment:
```suggestion
    )
    entry.async_on_unload(device.close)
```

Reply:
Ah cool, didn't know that this was possible!